integration/common/tests/sql/test_parser.py

```diff
@@ -382,3 +382,18 @@ def test_parse_multi_statement():
     )
     assert sql_meta.in_tables == [DbTableMeta('schema0.table0')]
     assert sql_meta.out_tables == [DbTableMeta('schema1.table1')]
+
+
+def test_parse_statement_list():
+    sql_meta = parse(["""
+        DROP TABLE IF EXISTS schema1.table1;
+        CREATE TABLE schema1.table1(
+          col0 VARCHAR(64),
+          col1 VARCHAR(64)
+        );""", """
+        INSERT INTO schema1.table1(col0, col1)
+          SELECT col0, col1
+            FROM schema0.table0;
+        """])
+    assert sql_meta.in_tables == [DbTableMeta('schema0.table0')]
```

Comment:
Shouldn't we treat this as a list of separete queries and return list of `sql_meta`?
Going this way can cause a mixup of the inputs & outputs of different queries. 
We also lose the information about the query ordering.

Reply:
Then, somewhere we need to "join" it as users of parser expect single result.

I think if you have distinct jobs and want to get distinct results, you can call parser in a loop

Reply:
yes, indeed.